src/ray/gcs/tables.cc

```diff
@@ -602,81 +628,78 @@ bool ClientTable::IsRemoved(const ClientID &node_id) const {
 Status ClientTable::Connect(const GcsNodeInfo &local_node_info) {
   RAY_CHECK(!disconnected_) << "Tried to reconnect a disconnected client.";
 
-  RAY_CHECK(local_node_info.node_id() == local_node_info_.node_id());
+  node_id_ = ClientID::FromBinary(local_node_info.node_id());
   local_node_info_ = local_node_info;
 
   // Construct the data to add to the client table.
   auto data = std::make_shared<GcsNodeInfo>(local_node_info_);
   data->set_state(GcsNodeInfo::ALIVE);
-  // Callback to handle our own successful connection once we've added
-  // ourselves.
-  auto add_callback = [this](RedisGcsClient *client, const UniqueID &log_key,
-                             const GcsNodeInfo &data) {
+  Status status = SyncAppend(JobID::Nil(), client_log_key_, data);
```

Comment:
This looks fine. Just wondering if this is mandatory to do a sync append?

Reply:
Yes.